scylla/src/network/connection.rs

```diff
@@ -952,24 +976,32 @@ impl Connection {
             .get_timestamp()
             .or_else(get_timestamp_from_gen);
 
+        let has_metadata_extension = self.features.protocol_features.scylla_metadata_id_supported;
+
+        // If the metadata id extension is supported, there is no point in not caching metadata,
+        // so we ignore the setting on prepared statement.
+        let skip_metadata =
+            prepared_statement.get_use_cached_result_metadata() || has_metadata_extension;
```

Comment:
:question: Once again: did you remember to update the docs on `PreparedStatement`?